File: routes/organization_routes.py
from fastapi import APIRouter
from controllers.organization_controller import OrganizationController

router = APIRouter(prefix="/organizations", tags=["Organizations"])

@router.post("/organization", response_model=dict, status_code=201)
async def create_organization(data: dict):
    """Create a new organization."""
    return await OrganizationController.create_organization(data)

@router.delete("/organization/{id}", response_model=dict)
async def delete_organization(id: str):
    return await OrganizationController.delete_organization(id)

@router.get("/organizations", response_model=list)
async def get_organizations():
    """Retrieve a list of organizations."""
    return await OrganizationController.get_organizations()

@router.get("/organization/{id}", response_model=dict)
async def get_organization_by_id(id: str):
    return await OrganizationController.get_organization_by_id(id)

@router.put("/organization/{id}", response_model=dict)
async def update_organization(id: str, data: dict):
    return await OrganizationController.update_organization(id, data)

File: routes/organizationRoutes.js
const express = require("express");
const router = express.Router();
const organizationController = require('../controllers/Organization')


/**
 * @swagger
 * /organization:
 *   post:
 *     summary: Create organization
 *     description: Creates a new organization.
 *     responses:
 *       201:
 *         description: Organization created successfully.
 */
router.post("/organization", organizationController.createOrganization);
router.delete('/organization/:id', organizationController.deleteOrganization)
/**
 * @swagger
 * /organizations:
 *   get:
 *     summary: Get organizations
 *     description: Retrieve a list of organizations.
 *     responses:
 *       200:
 *         description: Successfully retrieved list of organizations.
 */
router.get("/organizations", organizationController.getOrganizations);
router.get('/organization/:id', organizationController.getOrganizationBYId);
router.put('/organization/:id', organizationController.updateOrganization)

module.exports = router;